### ISSUE TITLE
List open buffers?

### ISSUE THREAD

**Issue by KevinSjoberg** [adsr/mle#23]
Is there a way to easily list open buffers or navigate to an already opened buffer? Whenever I edit multiple files (20+ buffers), I kind of get lost which buffer is which and have to manually do `M-n` and `M-p` to cycle through them one by one.

I assume it can be done using the Lua extension interface if not already implemented?

**Comment by adsr** [comment]
I have thought about adding a sidebar or topbar for this purpose however I was never able to come up with anything that felt useful enough to subtract screen real estate from edit buffers. Something similar to the file browser (`C-b`) would easy to implement. How do you envision it?

**Comment by KevinSjoberg** [comment]
Kind of tricky to find a sweet spot. I would not pursue the sidebar as it would impact the screen real estate.

I could potentially see FZF being used for this purpose allowing you to quickly navigate to an existing buffer.

Not sure I think that’s the ideal solution but it’s something. What do you think?

**Comment by adsr** [comment]
I added a buffer browser in 9853838c7856fd32a4724a64be26195a576f6865 which operates similar to the file browser.

I'll note that it is possible (albeit inefficient) to switch to an open buffer via fzf. The editor will avoid re-opening a file if it is already open and not unsaved[1]. (I fixed a bug yesterday concerning this behavior 46990ea3448dbc8cd54868dd03efdc22616cf26c.)

[1] To be honest I don't remember why I added that constraint. I am not sure it is the right behavior. I can't think of a reason why one would intentionally want the same file open twice in two different buffers.
